climpred/smoothing.py

```diff
@@ -0,0 +1,230 @@
+                return da
+            # elif 'lat_b' in ds.coords and 'lon_b' in ds.coords:
+            #    da = da.rename({'lat_b': 'lat', 'lon_b': 'lon'})
+            #    return da
+            elif 'TLAT' in ds.coords and 'TLONG' in ds.coords:
+                da = da.rename({'TLAT': 'lat', 'TLONG': 'lon'})
+                return da
+            else:
+                raise ValueError(
+                    'lon/lat or lon_b/lat_b or TLAT/TLON not found, please rename'
+                )
+
+        da = warn_lon_lat_dne(da)
+        grid_out = {
+            'lon': np.arange(da.lon.min(), da.lon.max() + d_lon, d_lon),
+            'lat': np.arange(da.lat.min(), da.lat.max() + d_lat, d_lat),
+        }
+        regridder = xe.Regridder(da, grid_out, **kwargs)
+        return regridder(da)
+
+    if 'lon' not in d_lon_lat_dict:
+        d_lon_lat_dict['lon'] = d_lon_lat_dict['lat']
+    elif 'lat' not in d_lon_lat_dict:
+        d_lon_lat_dict['lat'] = d_lon_lat_dict['lon']
+    else:
+        raise ValueError('please provide either lon or lat in d_lon_lat_dict.')
+
+    kwargs = {
+        'd_lon': d_lon_lat_dict['lon'],
+        'd_lat': d_lon_lat_dict['lat'],
+        'method': method,
+        'periodic': periodic,
+        'filename': filename,
+        'reuse_weights': reuse_weights,
+    }
+
+    ds = _regrid_it(ds, **kwargs)
+
+    return ds
+
+
+def spatial_smoothing_xrcoarsen(ds, coarsen_dict=None, how='mean'):
+    """Apply spatial smoothing by regridding to `boxsize` grid.
+
+    Reference:
+      * Goddard, L., A. Kumar, A. Solomon, D. Smith, G. Boer, P.
+            Gonzalez, V. Kharin, et al. “A Verification Framework for
+            Interannual-to-Decadal Predictions Experiments.” Climate
+            Dynamics 40, no. 1–2 (January 1, 2013): 245–72.
+            https://doi.org/10/f4jjvf.
+
+    Args:
+        ds (xr.object): input. xr.DataArray prefered.
+        coarsen_dict (dict): coarsen spatial latitudes.
+        how (str): aggregation type for coarsening. default: 'mean'
+
+    Returns:
+        ds_smoothed (xr.object): boxsize-regridded input
+
+    """
+    if coarsen_dict is None:
+        # guess spatial dims
+        spatial_dims_to_smooth = list(ds.dims)
+        for dim in ['time', 'lead', 'member', 'init']:
+            if dim in ds.dims:
+                spatial_dims_to_smooth.remove(dim)
+        # write coarsen to dict to coarsen similar to 5x5 degree
+        pass  # not implemented
+    # check whether coarsen dims are possible
+    for dim in coarsen_dict:
+        if dim not in ds.dims:
+            raise ValueError(dim, 'not in ds')
+        else:
+            if ds[dim].size % coarsen_dict[dim] != 0:
+                raise ValueError(
+                    coarsen_dict[dim], 'does not divide', ds[dim].size, 'in', dim
+                )
+    ds_out = getattr(ds.coarsen(coarsen_dict), how)()
+    return ds_out
+
+
+def temporal_smoothing(ds, smooth_dict={'time': 4}, how='mean', rename_dim=True):
```

Comment:
I think it's generally not good practice to have a dictionary/list/iterable as a default value because it can unforeseen results on multiple calls, but maybe just a list... I haven't tested with dict
https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html

Reply:
I see that now, but for the docstr I still want to put smooth_dict (dict).

Reply:
Yes that sounds good